DataManagementSystem/DB/FileCatalogComponents/DirectoryTreeBase.py

```diff
@@ -580,14 +585,17 @@ def getFileIDsInDirectory( self, dirID, credDict, startItem = 1, maxItems = 25 )
     return result
 
 
-  def getFilesInDirectory( self, dirID, credDict ):
+  def getFilesInDirectory( self, dirID, credDict, requestString = False ):
```

Comment:
This one is different. It is a simple one to just get the list of files explicitly or as a query without taking into account visibility logic, etc. But I agree that this should be rather moved to the File Manager implementation. All files must have a reference dirID but can be stored differently in different implementations. So, this would me more generic.